src/middleware/authUser.js

```diff
@@ -0,0 +1,36 @@
+const boom = require('boom')
+
+const roles = require('../constants/roles.js')
+const { auth } = require('../constants/errors.json')
+
+const User = require('../models/User')
+
+// function to check role of user matches minRole allowed on the route
+const verifyRole = ({ minRole }) => user => {
+  const userScope = user.scope
+
+  if (minRole === roles.BASIC) return
+  if (minRole === roles.ADMIN && userScope !== roles.BASIC) return
+  if (minRole === roles.SUPER && userScope === roles.SUPER) return
+
+  return Promise.reject(boom.unauthorized(auth.UNAUTHORIZED))
+}
+
+// Check user from the token exists, and add their id to the req.user for potential use in permissioning
+const addUserIdToSession = req => {
+  return User.find({ username: req.user.username }).then(users => {
+    if (users.length === 0) {
+      return Promise.reject(boom.unauthorized(auth.UNAUTHORIZED))
+    }
+
+    req.user.id = users[0].id
+    return req.user
+  })
+}
+
+module.exports = opts => (req, res, next) => {
+  addUserIdToSession(req)
```

Comment:
@des-des just getting things straight

ATM, we add their ID to the session, then check their role is allowed on this route, then let them through.

Does it not make more sense to check their role is allowed before adding their id? Not that it makes much difference but it seems weird this way.

Also, why do we not just add their id into the JWT, along with their username [here](https://github.com/foundersandcoders/open-tourism-platform/pull/95/files#diff-004d1e1bf2f9641418397529818e0943R50). I feel like we talked about this but now I can't remember.